libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java

```diff
@@ -190,44 +190,47 @@ public void onRestoreInstanceState(Bundle savedInstanceState) {
     mapInstanceState = savedInstanceState.getParcelable(MAP_INSTANCE_STATE_KEY);
   }
 
-  /**
-   * Called to ensure the {@link MapView} is destroyed
-   * properly.
-   * <p>
-   * In an {@link Activity} this should be in onDestroy().
-   * <p>
-   * In a {@link androidx.fragment.app.Fragment}, this should
-   * be in {@link androidx.fragment.app.Fragment#onDestroyView()}.
-   */
-  public void onDestroy() {
-    shutdown();
-    lifecycleRegistry.markState(Lifecycle.State.DESTROYED);
-  }
 
   public void onStart() {
     mapView.onStart();
     if (navigationMap != null) {
       navigationMap.onStart();
     }
-    lifecycleRegistry.markState(Lifecycle.State.STARTED);
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_START);
   }
 
   public void onResume() {
     mapView.onResume();
-    lifecycleRegistry.markState(Lifecycle.State.RESUMED);
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
   }
 
   public void onPause() {
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE);
```

Comment:
Noticed that we're normally (`onStart` and `onResume`) calling `lifecycleRegistry.handleLifecycleEvent` after `MapView`s lifecycle but not here 🤔 Is that on purpose? Could you clarify?

Reply:
sure, for this I left https://github.com/mapbox/mapbox-navigation-android/pull/3027#issuecomment-633531307